### PR TITLE
Eliminate unnecessary status lock acquisition in TP

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**:
Remove mutex acquisition when a thread updates its status to Spinning or Active.
This has no useful purpose.
Read atomics under the lock with relaxed semantics.

### CODE REVIEW

File: include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h [hunk 1]
```diff
@@ -1383,18 +1383,18 @@ class ThreadPoolTempl : public onnxruntime::concurrency::ExtendedThreadPoolInter
...
+      ThreadStatus seen = GetStatus();
       if (seen == ThreadStatus::Blocking ||
           seen == ThreadStatus::Blocked) {
         std::unique_lock<OrtMutex> lk(mutex);
         // Blocking state exists only transiently during the SetBlock() method
         // while holding the lock.  We may observe it at the start of this
         // function, but after acquiring the lock then the target thread
         // will either be blocked or not.
-        seen = status;
+        seen = status.load(std::memory_order_relaxed);
```
Comment (pranavsharma) [inline include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h:1394]: Would be good to comment why this change is being made.

**Review by pranavsharma — DISMISSED** [review]

**Review by tlh20 — DISMISSED** [review]

**Review by pranavsharma — APPROVED** [review]

**Comment by pranavsharma** [comment]
Curious if this brought any latency improvements. If yes, would be good to record them here for reference.
